### DISCUSSION TITLE
Pruning: Export ONNX with pruned connections removed

### DISCUSSION THREAD

**Discussion by saravanabalagi** [open-mmlab/mmrazor#588] [Q&A]
Say I do filter pruning, does mmrazor change the architecture or retain a prune mask which gets applied when model forward is called similar to the one in torch.nn.utils.prune?

When I export ONNX will the pruned connections be removed completely? i.e., I am not looking for weights filled with zeros making it sparse, instead I want the filters/channels/neurons to be removed completely from the network. Is this possible using mmrazor?

**Comment by iammuhammadamir** [comment]
Hey @saravanabalagi , 
Did you get answer to your question ? Kindly share with me. 
I have actually pruned the model using ItePruneAlgorithm from MMRazor. Now, IDK how to export it.
